src/Cli/Exporter.cs

```diff
@@ -115,27 +118,66 @@ private static async Task ExportGraphQL(ExportOptions options, RuntimeConfig run
             logger.LogInformation("Schema file exported successfully at {0}", options.OutputDirectory);
         }
 
-        private static string ExportGraphQLFromDabService(RuntimeConfig runtimeConfig, ILogger logger)
+        /// <summary>
+        /// Fetches the GraphQL schema from the DAB service, attempting to use the HTTPS endpoint first.
+        /// If the HTTPS endpoint fails, it falls back to using the HTTP endpoint.
+        /// Logs the process of fetching the schema and any fallback actions taken.
+        /// </summary>
+        /// <param name="runtimeConfig">The runtime config object containing the GraphQL path.</param>
+        /// <param name="logger">The logger instance used to log information and errors during the schema fetching process.</param>
+        /// <returns>The GraphQL schema as a string.</returns>
+        internal string ExportGraphQLFromDabService(RuntimeConfig runtimeConfig, ILogger logger)
         {
             string schemaText;
             // Fetch the schema from the GraphQL API
             logger.LogInformation("Fetching schema from GraphQL API.");
 
-            HttpClient client = new( // CodeQL[SM02185] Loading internal server connection
+            try
+            {
+                logger.LogInformation("Trying to fetch schema from DAB Service using HTTPS endpoint.");
+                schemaText = GetGraphQLSchema(runtimeConfig, useFallbackURL: false);
+            }
+            catch
+            {
+                logger.LogInformation("Failed to fetch schema from DAB Service using HTTPS endpoint. Trying with HTTP endpoint.");
```

Comment:
What happens if the schema fetch fails even from the fallback option i.e. http

Reply:
it will give failure with log indicating we tried HTTPS and then HTTP and then failed to export grpahQL